clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2441,16 +2486,26 @@ scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input_as_previously(
     return sched_type_t::STATUS_OK;
 }
 
+template <typename RecordType, typename ReaderType>
+std::unique_lock<std::mutex>
+scheduler_tmpl_t<RecordType, ReaderType>::acquire_scoped_sched_lock_if_necessary(
+    bool &need_lock)
+{
+    need_lock =
+        options_.mapping == MAP_TO_ANY_OUTPUT || options_.mapping == MAP_AS_PREVIOUSLY;
```

Comment:
Trying to find where it is repeated: this function is already a common routine I created to avoid repeating.  I do see one other case, passing hold_sched_lock to eof_or_idle on line 2368.  Is that one case what you mean?

Reply:
I made need_sched_lock() for these 2 uses.